01_classify_stratified/run_mutation_classification.py

```diff
@@ -22,57 +22,95 @@
+    opts.add_argument('--debug', action='store_true',
+                      help='use subset of data for fast debugging')
+    opts.add_argument('--num_folds', type=int, default=4,
+                      help='number of folds of cross-validation to run')
+    opts.add_argument('--seed', type=int, default=cfg.default_seed)
+    opts.add_argument('--subset_mad_genes', type=int, default=cfg.num_features_raw,
+                      help='if included, subset gene features to this number of '
+                           'features having highest mean absolute deviation')
+    opts.add_argument('--training_data', type=str, default='expression',
+                      choices=['expression', 'methylation'],
+                      help='what data type to train model on')
+
+    args = parser.parse_args()
+
+    args.results_dir = Path(args.results_dir).resolve()
+
+    if args.log_file is None:
+        args.log_file = Path(args.results_dir, 'log_skipped.tsv').resolve()
 
     if args.gene_set == 'custom':
         if args.custom_genes is None:
-            p.error('must include --custom_genes when --gene_set=\'custom\'')
+            parser.error('must include --custom_genes when --gene_set=\'custom\'')
         args.gene_set = args.custom_genes
         del args.custom_genes
     elif (args.gene_set != 'custom' and args.custom_genes is not None):
-        p.error('must use option --gene_set=\'custom\' if custom genes are included')
+        parser.error('must use option --gene_set=\'custom\' if custom genes are included')
 
-    args.results_dir = Path(args.results_dir).resolve()
+    # split args into defined argument groups, since we'll use them differently
+    arg_groups = du.split_argument_groups(args, parser)
+    io_args, model_options = arg_groups['io'], arg_groups['model_options']
 
-    if args.log_file is None:
-        args.log_file = Path(args.results_dir, 'log_skipped.tsv').resolve()
+    # add some additional hyperparameters/ranges from config file to model options
+    model_options.alphas = cfg.alphas
+    model_options.l1_ratios = cfg.l1_ratios
+    model_options.standardize_data_types = cfg.standardize_data_types
 
-    return args
+    return io_args, model_options
 
 
 if __name__ == '__main__':
 
     # process command line arguments
-    args = process_args()
-    sample_info_df = du.load_sample_info(verbose=args.verbose)
+    io_args, model_options = process_args()
+    sample_info_df = du.load_sample_info(verbose=io_args.verbose)
 
     # create results dir if it doesn't exist
-    args.results_dir.mkdir(parents=True, exist_ok=True)
+    io_args.results_dir.mkdir(parents=True, exist_ok=True)
+    # create subdir for experiment if it doesn't exist
+    experiment_dir = Path(io_args.results_dir, 'gene').resolve()
+    experiment_dir.mkdir(parents=True, exist_ok=True)
+
+    # save model options for this experiment
+    # (hyperparameters, preprocessing info, etc)
+    fu.save_model_options(experiment_dir, model_options)
```

Comment:
Just thinking aloud (minor comment): would it be helpful to have this function in `TCGADataModel` instead?

Reply:
Yeah, that might make more sense. I'm going to leave it for now, but I'll make a note in #11 to think about this in the future (since I'll be refactoring the argument parsing/saving code anyway).